examples/drawing/plot_mpl_annotations.py

```diff
@@ -0,0 +1,72 @@
+"""
+=================
+Annotations with Matplotlib
+=================
+
+Example of using matplotlib to place annotations on top of a network
+"""
+
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import networkx as nx
+import urllib.request
+import io
+
+# Image URLs for graph nodes
+URLs = [
+    {
+        "type": "router",
+        "URL": "https://www.materialui.co/materialIcons/hardware/router_black_144x144.png",
+    },
+    {
+        "type": "switch",
+        "URL": "https://www.materialui.co/materialIcons/action/dns_black_144x144.png",
+    },
+    {
+        "type": "PC",
+        "URL": "https://www.materialui.co/materialIcons/hardware/computer_black_144x144.png",
+    },
+]
+
+# Import images from URLS into a collection
+images = {}
+for u in URLs:
+    with urllib.request.urlopen(u["URL"]) as url:
+        image = mpimg.imread(io.BytesIO(url.read()))
+        images[u["type"]] = image
```

Comment:
Of course, there's no real need to even have the two separate dictionaries, you could just combine the calls into one, though that might be more difficult to read since the URLs are so long - whatever you think!